crates/ruff_python_ast/src/visitor.rs

```diff
@@ -573,11 +573,11 @@ pub fn walk_format_spec<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, format_spe
 }
 
 pub fn walk_arguments<'a, V: Visitor<'a> + ?Sized>(visitor: &mut V, arguments: &'a Arguments) {
-    for arg in &arguments.args {
-        visitor.visit_expr(arg);
-    }
-    for keyword in &arguments.keywords {
-        visitor.visit_keyword(keyword);
+    for arg_or_keyword in arguments.args_and_keywords() {
```

Comment:
The ordering in this visitor should be the same as the execution order at runtime.

Reply:
Input:
```python
def f(*args, **kwargs):
    pass

def g(x):
    print(x)
    return x


f(*g([1]), a=g(2), *g([3]), **g({"4": 5}))
```
Output:
```python
[1]
[3]
2
{'4': 5}
```
:face_with_spiral_eyes:

Reply:
thanks for making me check